Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -858,6 +873,38 @@ public void RaiseFocusExit(IMixedRealityPointer pointer, GameObject unfocusedObj
 
         #region Pointers
 
+        /// <inheritdoc />
+        public async void RegisterPointerHandler(IMixedRealityPointerHandler handler)
+        {
+            if (pointerHandlers == null)
+            {
+                pointerHandlers = new HashSet<IMixedRealityPointerHandler>();
+            }
+            // We do this check to guard against changes to the pointer handler collection while we're iterating over it.
+            else if (eventExecutionDepth > 0)
+            {
+                await doneExecutingEvents;
+            }
```

Comment:
If we wanted to be really pendantic here should should probably be its own if, so that changes to the pointerHandlers is also done in a way that is ordered after all event execution. I think the weirdness comes from how we're using doneExecutingEvents and eventExecutionDepth, which were designed to protect a different data structure (i.e. ideally this would be pointerEventExecutionDepth since it doesn't matter if there are other global events firing).

Anyways, just thinking out loud. I think in the end this doesn't matter and I think it's fine as is.